src/Symfony/Component/Validator/Tests/Mapping/Cache/AbstractCacheTest.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Tests\Mapping\Cache;
+
+use Symfony\Component\Validator\Mapping\Cache\CacheInterface;
+
+abstract class AbstractCacheTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @var CacheInterface
+     */
+    protected $cache;
+
+    public function testWrite()
+    {
+        $meta = $this->getMockBuilder('Symfony\\Component\\Validator\\Mapping\\ClassMetadata')
+            ->disableOriginalConstructor()
+            ->setMethods(array('getClassName'))
+            ->getMock();
+
+        $meta->expects($this->once())
+            ->method('getClassName')
+            ->will($this->returnValue('bar'));
```

Comment:
I would pick an actual classname with backslashes maybe? Then tests would fail if your `escapeClassName` method would not escape properly ;)